public/app/features/explore/RichHistory/RichHistoryAddToLibrary.tsx

```diff
@@ -1,57 +1,32 @@
 import { useState } from 'react';
 
 import { DataQuery } from '@grafana/schema';
-import { Button, Modal } from '@grafana/ui';
+import { Button } from '@grafana/ui';
 import { t } from 'app/core/internationalization';
-import { isQueryLibraryEnabled, useListQueryTemplateQuery } from 'app/features/query-library';
 
-import {
-  queryLibraryTrackAddFromQueryHistory,
-  queryLibraryTrackAddFromQueryHistoryAddModalShown,
-} from '../QueryLibrary/QueryLibraryAnalyticsEvents';
-import { QueryTemplateForm } from '../QueryLibrary/QueryTemplateForm';
+import { useQueryLibraryContext } from '../QueryLibrary/QueryLibraryContext';
 
 type Props = {
   query: DataQuery;
 };
 
 export const RichHistoryAddToLibrary = ({ query }: Props) => {
-  const { refetch } = useListQueryTemplateQuery({});
-  const [isOpen, setIsOpen] = useState(false);
   const [hasBeenSaved, setHasBeenSaved] = useState(false);
+  const { openAddQueryModal, queryLibraryEnabled } = useQueryLibraryContext();
 
   const buttonLabel = t('explore.rich-history-card.add-to-library', 'Add to library');
 
-  return isQueryLibraryEnabled() && !hasBeenSaved ? (
+  return queryLibraryEnabled && !hasBeenSaved ? (
     <>
       <Button
         variant="secondary"
         aria-label={buttonLabel}
         onClick={() => {
-          setIsOpen(true);
-          queryLibraryTrackAddFromQueryHistoryAddModalShown();
+          openAddQueryModal(query, { onSave: () => setHasBeenSaved(true), context: 'richHistory' });
```

Comment:
This had duplicated code for the modal using the internal components of query library. This call to open modal from the context should be equivalent.